### PR TITLE
hack: [API28] fix crash on 'Filtered Deck Options'

### LINKED ISSUE #15536: [Bug] API 28: Can't open Filtered Deck Options
* make a custom study deck
* Long press - Deck Options

`android.content.res.Resources$NotFoundException: File res/color/faded_primary.xml from drawable resource ID #0x7f06007d`

`Binary XML file line #17: <item> tag requires a 'drawable' attribute or child tag defining a drawable`

```
ACRA caught a RuntimeException for com.ichi2.anki.debug
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki.debug/com.ichi2.anki.FilteredDeckOptions}: android.view.InflateException: Binary XML file line #27: Binary XML file line #27: Error inflating class androidx.appcompat.widget.ActionBarContainer
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2707)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1460)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6077)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
Caused by: android.view.InflateException: Binary XML file line #27: Binary XML file line #27: Error inflating class androidx.appcompat.widget.ActionBarContainer
Caused by: android.view.InflateException: Binary XML file line #27: Error inflating class androidx.appcompat.widget.ActionBarContainer
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:430)
	at android.view.LayoutInflater.createView(LayoutInflater.java:645)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:787)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:858)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:518)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:377)
	at androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor(AppCompatDelegateImpl.java:979)
	at androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor(AppCompatDelegateImpl.java:889)
	at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:772)
	at com.ichi2.ui.AppCompatPreferenceActivity.setContentView(AppCompatPreferenceActivity.kt:211)
	at android.preference.PreferenceActivity.onCreate(PreferenceActivity.java:557)
	at com.ichi2.ui.AppCompatPreferenceActivity.onCreate(AppCompatPreferenceActivity.kt:185)
	at com.ichi2.anki.FilteredDeckOptions.onCreate(FilteredDeckOptions.kt:200)
	at android.app.Activity.performCreate(Activity.java:6662)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2599)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2707)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1460)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6077)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
Caused by: android.content.res.Resources$NotFoundException: Drawable com.ichi2.anki.debug:color/faded_primary with resource ID #0x7f06007d
Caused by: android.content.res.Resources$NotFoundException: File res/color/faded_primary.xml from drawable resource ID #0x7f06007d
	at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:725)
	at android.content.res.ResourcesImpl.loadDrawable(ResourcesImpl.java:571)
	at android.content.res.Resources.loadDrawable(Resources.java:854)
	at android.content.res.TypedArray.getDrawable(TypedArray.java:928)
	at androidx.appcompat.widget.ActionBarContainer.<init>(ActionBarContainer.java:68)
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:430)
	at android.view.LayoutInflater.createView(LayoutInflater.java:645)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:787)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:858)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:518)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:377)
	at androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor(AppCompatDelegateImpl.java:979)
	at androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor(AppCompatDelegateImpl.java:889)
	at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:772)
	at com.ichi2.ui.AppCompatPreferenceActivity.setContentView(AppCompatPreferenceActivity.kt:211)
	at android.preference.PreferenceActivity.onCreate(PreferenceActivity.java:557)
	at com.ichi2.ui.AppCompatPreferenceActivity.onCreate(AppCompatPreferenceActivity.kt:185)
	at com.ichi2.anki.FilteredDeckOptions.onCreate(FilteredDeckOptions.kt:200)
	at android.app.Activity.performCreate(Activity.java:6662)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2599)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2707)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1460)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6077)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #17: <item> tag requires a 'drawable' attribute or child tag defining a drawable
	at android.graphics.drawable.StateListDrawable.inflateChildElements(StateListDrawable.java:183)
	at android.graphics.drawable.StateListDrawable.inflate(StateListDrawable.java:116)
	at android.graphics.drawable.DrawableInflater.inflateFromXml(DrawableInflater.java:130)
	at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:1227)
	at android.graphics.drawable.Drawable.createFromXml(Drawable.java:1200)
	at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:715)
	... 33 more
```

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
`android.content.res.Resources$NotFoundException: File res/color/faded_primary.xml from drawable resource ID #0x7f06007d`

`Binary XML file line https://github.com/ankidroid/Anki-Android/pull/17: <item> tag requires a 'drawable' attribute or child tag defining a drawable`

## Fixes
* Fixes #15536

## Approach
We fix the XML issue by hardcoding the colors

0.06 * 256 = 15.36, so use an alpha of 15/256 (#0FXXXXXX)

## How Has This Been Tested?
API 24 emulator

Samsung S21 (API 34)
<details>

<summary>Before</summary>

image

image

</details>

<details>

<summary>After</summary>

image

image

</details>

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by mikehardy — COMMENTED** [review]
isn't the way to do this to keep faded_primary.xml, but then have a values-29 folder where you use it, and only in the mainline do you use the hard-coded color?

Then similar to the compat infrastructure as we bump minSdk there's a natural queue to revert things?

Also, I see a scattering of API 24, 26, 28, and 29 mentioned. Which is it?
Might be nice to focus on API28 only (I think that's the one with the bug?) and say in comments "API28 workaround, revert when > 28" or similar so searches for API numbers only hit on the one

**Comment by david-allison** [comment]
* Fails: API 28
* Passes: API 29

**Comment by david-allison** [comment]
* Removed a reference to API 26

We want to test on the lowest possible API. The problem exists in API 24 - 28, so testing on 24 feels reasonable to ensure there aren't other regressions

File: AnkiDroid/src/main/res/values/attrs.xml [hunk 1]
```diff
@@ -167,4 +167,7 @@
     <attr name="checkMediaListForeground" format="color" />
 
     <attr name="progressDialogButtonTextColor" format="color"/>
+
+    <!-- API 28 workaround. Revert when > APU 28 -->
```
Comment (mikehardy) [inline AnkiDroid/src/main/res/values/attrs.xml:171]:
```suggestion
    <!-- API 28 workaround. Revert when > API 28 -->
```

**Review by mikehardy — APPROVED** [review]
alright, that's a super searchable string, and this is easier than the values-API construct (I kind of like that but it's also pretty heavily engineered) so this works for me

I pointed out one typo which I don't even *really* care about and I think you've already applied it because I see "Refresh" while typing this :-)

**Comment by mikehardy** [comment]
I still see the typo in there - if you want to fix it, push a fix, but in general merge when ready whenever you want to click the button IMHO

File: AnkiDroid/src/main/res/values/attrs.xml (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/main/res/values/attrs.xml:171]: image

**Comment by BrayanDSO** [comment]
Here's the patch of what Mike wanted to do:
```diff
Subject: [PATCH] a
---
Index: AnkiDroid/src/main/res/values-v29/theme_light.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/res/values-v29/theme_light.xml b/AnkiDroid/src/main/res/values-v29/theme_light.xml
new file mode 100644
--- /dev/null	(revision ff4da484c86892b95280b32ee6bceb0c4e19be61)
+++ b/AnkiDroid/src/main/res/values-v29/theme_light.xml	(revision ff4da484c86892b95280b32ee6bceb0c4e19be61)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources xmlns:tools="http://schemas.android.com/tools" tools:ignore="DuplicateCrowdInStrings">
+    <style name="Base.V29.Light" parent="@style/Base.Theme.Light">
+        <item name="colorSurfaceContainer">@color/faded_primary</item>
+    </style>
+
+    <style name="Theme_Light" parent="Base.V29.Light"/>
+</resources>
Index: AnkiDroid/src/main/res/values/theme_light.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/res/values/theme_light.xml b/AnkiDroid/src/main/res/values/theme_light.xml
--- a/AnkiDroid/src/main/res/values/theme_light.xml	(revision 6d2f895133b8b7aab4e5dad1f335e07e778db5e5)
+++ b/AnkiDroid/src/main/res/values/theme_light.xml	(revision ff4da484c86892b95280b32ee6bceb0c4e19be61)
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 
 <resources xmlns:tools="http://schemas.android.com/tools" tools:ignore="DuplicateCrowdInStrings">
-    <style name="Theme_Light" parent="@style/Theme.Material3.Light.NoActionBar">
+    <style name="Base.Theme.Light" parent="@style/Theme.Material3.Light.NoActionBar">
         <!-- Android colors -->
         <item name="colorPrimary">@color/material_light_blue_500</item>
         <item name="colorPrimaryDark">@color/material_light_blue_700</item>
@@ -13,7 +13,7 @@
 
         <item name="colorSurfaceContainerHighest">?attr/colorSurfaceContainer</item>
         <item name="colorSurfaceContainerHigh">?android:attr/colorBackground</item>
-        <item name="colorSurfaceContainer">@color/faded_primary</item>
+        <item name="colorSurfaceContainer">#0F03A9F4</item>
         <item name="colorSurfaceContainerLow">?android:attr/colorBackground</item>
 
         <item name="android:textColor">@color/text_color_light</item>
@@ -141,6 +141,7 @@
         <item name="checkMediaListBackground">@color/material_blue_grey_050</item>
         <item name="checkMediaListForeground">@color/material_grey_700</item>
         <item name="progressDialogButtonTextColor">@color/material_light_blue_500</item>
-
     </style>
+
+    <style name="Theme_Light" parent="Base.Theme.Light"/>
 </resources>
```

**Comment by BrayanDSO** [comment]
Also, the IDE shows me this 
image

**Comment by david-allison** [comment]
@BrayanDSO I have no preference, would you prefer this?

**Comment by mikehardy** [comment]
I'm on the fence personally. It is a way of externalizing the ability to search on API things you can compress later as minSdk moves. But we do searches already 🤷 . You are the deciding vote @BrayanDSO 😄

**Comment by BrayanDSO** [comment]
> @BrayanDSO I have no preference, would you prefer this?

yes, using `-v29` resources is the appropriate way of handling things like this.

**Comment by mikehardy** [comment]
there it is. and you're not wrong

**Comment by BrayanDSO** [comment]
```diff
Subject: [PATCH] a
---
Index: AnkiDroid/src/main/res/values-v29/theme_light.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/res/values-v29/theme_light.xml b/AnkiDroid/src/main/res/values-v29/theme_light.xml
new file mode 100644
--- /dev/null	(date 1708216811737)
+++ b/AnkiDroid/src/main/res/values-v29/theme_light.xml	(date 1708216811737)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources xmlns:tools="http://schemas.android.com/tools" tools:ignore="DuplicateCrowdInStrings">
+    <style name="Base.V29.Light" parent="@style/Base.Theme.Light">
+        <item name="colorSurfaceContainer">@color/faded_primary</item>
+    </style>
+
+    <style name="Theme_Light" parent="Base.V29.Light"/>
+</resources>
Index: AnkiDroid/src/main/res/values-v29/theme_dark.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/res/values-v29/theme_dark.xml b/AnkiDroid/src/main/res/values-v29/theme_dark.xml
new file mode 100644
--- /dev/null	(date 1708217453320)
+++ b/AnkiDroid/src/main/res/values-v29/theme_dark.xml	(date 1708217453320)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources xmlns:tools="http://schemas.android.com/tools" tools:ignore="DuplicateCrowdInStrings">
+    <style name="Base.V29.Dark" parent="@style/Base.Theme.Dark">
+        <item name="colorSurfaceContainer">@color/faded_primary</item>
+    </style>
+
+    <style name="Theme_Dark" parent="Base.V29.Dark"/>
+</resources>
Index: AnkiDroid/src/main/res/values/theme_light.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/res/values/theme_light.xml b/AnkiDroid/src/main/res/values/theme_light.xml
--- a/AnkiDroid/src/main/res/values/theme_light.xml	(revision 6d2f895133b8b7aab4e5dad1f335e07e778db5e5)
+++ b/AnkiDroid/src/main/res/values/theme_light.xml	(date 1708217018849)
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 
 <resources xmlns:tools="http://schemas.android.com/tools" tools:ignore="DuplicateCrowdInStrings">
-    <style name="Theme_Light" parent="@style/Theme.Material3.Light.NoActionBar">
+    <style name="Base.Theme.Light" parent="@style/Theme.Material3.Light.NoActionBar">
         <!-- Android colors -->
         <item name="colorPrimary">@color/material_light_blue_500</item>
         <item name="colorPrimaryDark">@color/material_light_blue_700</item>
@@ -13,7 +13,7 @@
 
         <item name="colorSurfaceContainerHighest">?attr/colorSurfaceContainer</item>
         <item name="colorSurfaceContainerHigh">?android:attr/colorBackground</item>
-        <item name="colorSurfaceContainer">@color/faded_primary</item>
+        <item name="colorSurfaceContainer">#0F03A9F4</item>
         <item name="colorSurfaceContainerLow">?android:attr/colorBackground</item>
 
         <item name="android:textColor">@color/text_color_light</item>
@@ -141,6 +141,7 @@
         <item name="checkMediaListBackground">@color/material_blue_grey_050</item>
         <item name="checkMediaListForeground">@color/material_grey_700</item>
         <item name="progressDialogButtonTextColor">@color/material_light_blue_500</item>
-
     </style>
+
+    <style name="Theme_Light" parent="Base.Theme.Light"/>
 </resources>
Index: AnkiDroid/src/main/res/values/theme_dark.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/res/values/theme_dark.xml b/AnkiDroid/src/main/res/values/theme_dark.xml
--- a/AnkiDroid/src/main/res/values/theme_dark.xml	(revision 6d2f895133b8b7aab4e5dad1f335e07e778db5e5)
+++ b/AnkiDroid/src/main/res/values/theme_dark.xml	(date 1708217453316)
@@ -9,7 +9,7 @@
     <color name="theme_dark_drawer_row_current">#33999999</color>
     <color name="theme_dark_drawer_item">#DEFFFFFF</color>
 
-    <style name="Theme_Dark" parent="@style/Theme.Material3.Dark.NoActionBar">
+    <style name="Base.Theme.Dark" parent="@style/Theme.Material3.Dark.NoActionBar">
         <!-- Android colors -->
         <item name="colorPrimary">@color/material_blue_400</item>
         <item name="colorOnPrimary">@color/white</item>
@@ -27,7 +27,7 @@
 
         <item name="colorSurfaceContainerHighest">?attr/colorSurfaceContainer</item>
         <item name="colorSurfaceContainerHigh">?android:attr/colorBackground</item>
-        <item name="colorSurfaceContainer">@color/faded_primary</item>
+        <item name="colorSurfaceContainer">#0F03A9F4</item>
         <item name="colorSurfaceContainerLow">?android:attr/colorBackground</item>
         <!-- Widget colors -->
         <item name="widgetColorActivated">?attr/colorAccent</item>
@@ -155,4 +155,6 @@
         <item name="alertDialogTheme">@style/AlertDialogStyle</item>
 
     </style>
+
+    <style name="Theme_Dark" parent="Base.Theme.Dark"/>
 </resources>
```

**Comment by BrayanDSO** [comment]
as an issue, plain theme needs a color that isn't blue

**Comment by david-allison** [comment]
Tested new code on API 28 and 34. I did the dark theme by myself

**Comment by david-allison** [comment]
image

Thanks for the help, never enjoyed Android theming

**Comment by BrayanDSO** [comment]
> Thanks for the help, never enjoyed Android theming

Neither I have

ItsJustWhatIHaveToLiveWithJenniFarleyGIF

**Comment by mikehardy** [comment]
you two theming

image

**Review by mikehardy — APPROVED** [review]
Just scanned the final version FWIW, looks good

**Pull request merged**